config/dba.neon

```diff
@@ -33,6 +33,7 @@ services:
                 - 'Doctrine\DBAL\Connection::iterateAssociativeIndexed'
                 - 'Doctrine\DBAL\Connection::iterateColumn'
                 - 'Doctrine\DBAL\Connection::executeUpdate' # deprecated in doctrine
+                - 'Dibi\Connection::fetch'
```

Comment:
please rebase your branch and move the configs into `rules.neon`.

also please add test-cases into `SyntaxErrorInPreparedStatementMethodRuleTest`, see
- `syntax-error-in-prepared-statement.php`
- `syntax-error-with-inference-placeholder.php`

which shows that the rule is detecting syntax errors in queries given to `Dibi\Connection::fetch`.
also test a case which shows that queries without errors don't get errors reported.

Reply:
> please rebase your branch and move the configs into the newly added rules.neon.

done

> also please add test-cases into SyntaxErrorInPreparedStatementMethodRuleTest, see

not really sure how to add tests there. There's comparison for the reflector, eg `if (MysqliQueryReflector::NAME === getenv('DBA_REFLECTOR')) {` but you said that DibiQueryReflector shouldnt be part of this PR. Can you please explain further?

Reply:
if I got it right, Dibi supports a mysql backend?
I did not check whether it also supports postgres.
these are the 2 only backend phpstan-dba atm supports

if I am right, for the syntax query check the dibi method will just use one of the existing mysql backend connectors. we don't need to run the queries thru bidi itself at analysis time.

Reply:
> if I got it right, Dibi supports a mysql backend?

that is correct. Nevertheless, I am still confused as to what to do with the tests :(

Reply:
I will add a single test so you get an idea.